imslp.js

```diff
@@ -0,0 +1,239 @@
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (getSearchResults(doc, true)) {
+		return 'multiple';
+	}
+	else if (url.includes('/wiki/')) {
+		return 'book';
+	}
+	return false;
+}
+
+function getSearchResults(doc, url, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('.categorypagelink[href]');
+	for (var i = 0; i < rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (!items) return;
+		for (var urls of Object.keys(items)) {
+			await scrape(await requestDocument(urls));
+		}
+		//Z.debug(ZU.processDocuments(Object.keys(items)), scrape)
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+async function scrape(doc, url = doc.location.href) {
+	var item = new Zotero.Item("book");
+	var title = text(doc, '#firstHeading');
+	
+	//check metadata
+	var rows = doc.querySelector('.wi_body table').rows;
+	for (var i = 2; i < rows.length; i++) {
+		//Z.debug(rows[i].cells[0].innerText);
+		//Z.debug(rows[i].cells[1].innerText);
+		if (rows[i].cells[0].innerText.trim().includes("Name Translations")) {
+			var names = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim() == "Composer") {
+			var author = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim().includes("Opus/Catalogue Number")) {
+			var seriesNumber = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim() == "Key") {
+			var key = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim() == "Text Incipit") {
+			var incipit = rows[i].cells[1].innerHTML;
+		}
+		if (rows[i].cells[0].innerText.trim().includes("Movements/Sections")) {
+			var movements = rows[i].cells[1].innerHTML;
+		}
+		if (rows[i].cells[0].innerText.trim().includes("Year/Date of Composition")) {
+			var date = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim().includes("First Performance")) {
+			var place = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim().includes("First Publication")) {
```

Comment:
I don't think this makes sense -- if we want the publisher, we'd need to grab it from the scores above, not use the data of first plublication, i.e. Vienna: Universal_Edition

Reply:
A score may published bu many publishers... so just removed it.

Reply:
But at that point, what are we left with? If we don't get publisher, then place of publication doesn't make sense, publication date (which is specific to a score) doesn't make sense, we can't include editor (which, again, is specific to an edition), so you're basically just left with composer and title. Is that really worth a translator?